BitwardenShared/UI/Vault/VaultItem/ViewItem/ViewLoginItem/Extensions/CipherView+Update.swift

```diff
@@ -184,21 +194,56 @@ extension CipherView {
             viewPassword: viewPassword,
             localData: localData,
             attachments: attachments,
-            fields: addEditState.customFieldsState.customFields.isEmpty ?
-                nil : addEditState.customFieldsState.customFields.map { customField in
-                    FieldView(
-                        name: customField.name,
-                        value: customField.value,
-                        type: .init(fieldType: customField.type),
-                        linkedId: customField.linkedIdType?.rawValue
-                    )
-                },
+            fields: newFields,
             passwordHistory: passwordHistory,
             creationDate: creationDate,
             deletedDate: deletedDate,
             revisionDate: revisionDate
         )
     }
+
+    private func updatePasswordHistoryWithHiddenFields(
+        passwordHistory: [PasswordHistoryView]?,
+        oldFields: [FieldView]?,
+        newFields: [FieldView]?,
+        lastUsedDate: Date
+    ) -> [PasswordHistoryView]? {
+        guard let oldFields else {
+            return passwordHistory
+        }
+
+        var newHistory = passwordHistory
+        oldFields.filter { field in
+            FieldType(fieldType: field.type) == FieldType.hidden &&
+                !field.name.isEmptyOrNil &&
+                !field.value.isEmptyOrNil &&
+                !(newFields?.contains(field) ?? false)
+        }.forEach { hiddenField in
+            guard !hiddenField.name.isEmptyOrNil else {
+                return
+            }
+
+            let passHistoryHiddenField = PasswordHistoryView(
+                password: "\(hiddenField.name ?? ""): \(hiddenField.value ?? "")",
+                lastUsedDate: lastUsedDate
+            )
+            if newHistory == nil {
+                newHistory = [passHistoryHiddenField]
+            } else {
+                newHistory?.append(passHistoryHiddenField)
+            }
+        }
+        return newHistory
```

Comment:
🎨 I was thinking about doing this change to improve the code and avoid having the extra logic to create the array if the `newHistory == nil`. What do you think about this? Do you feel like it's easier to read or your approach is? I feel like I have pros and cons on both approaches but leaning more towards this one but would love to hear your thoughts on this as well. 
```suggestion
        let newPasswordHistoryFields = oldFields
            .filter { field in
                FieldType(fieldType: field.type) == FieldType.hidden &&
                    !field.name.isEmptyOrNil &&
                    !field.value.isEmptyOrNil &&
                    !(newFields?.contains(field) ?? false)
            }.compactMap { hiddenField in
                guard !hiddenField.name.isEmptyOrNil else {
                    return nil
                }

                return PasswordHistoryView(
                    password: "\(hiddenField.name ?? ""): \(hiddenField.value ?? "")",
                    lastUsedDate: lastUsedDate
                )
            }
        guard !newPasswordHistoryFields.isEmpty else {
            return passwordHistory
        }
        return passwordHistory?.append(contentsOf: newPasswordHistoryFields) ?? newPasswordHistoryFields
```